tests/services/storage/test_auth.py

```diff
@@ -28,37 +31,52 @@ def mock_tmp_auth_conn(mock_connection, py42_response):
 
 
 @pytest.fixture
-def mock_v1_auth_conn(mock_connection, py42_response):
-    py42_response.text = '["TEST_V1", "TOKEN_VALUE"]'
-    mock_connection.post.return_value = py42_response
-    return mock_connection
+def mock_storage_auth_token_conn(mocker):
+    response = mocker.MagicMock(spec=Py42Response)
+    response.status_code = 200
+    response.encoding = None
+    response.__getitem__ = lambda _, key: json.loads(response.text)[key]
+    response.text = '["TEST_V1", "TOKEN_VALUE"]'
```

Comment:
It's much easier to just mock the `requests.Request` and use a legit `Py42Response`, from my experience this saves even future headaches.

Then, all you need to do is set the `.text` on the mock response and let the normal `Py42Response` handle the rest.

Most of the mock response in py42 are currently like this

Reply:
Ah, good to know. I was just copying the fixture code from conftest since re-using the same fixtures in the same test seemed to be causing them to overwrite each other.